src/vs/workbench/api/common/extHostNotebookDocuments.ts

```diff
@@ -13,6 +12,35 @@ import { NotebookDocumentMetadata } from 'vs/workbench/contrib/notebook/common/n
 import { SerializableObjectWithBuffers } from 'vs/workbench/services/extensions/common/proxyIdentifier';
 import type * as vscode from 'vscode';
 
+declare const Buffer: any;
+
+const hasBuffer = (typeof Buffer !== 'undefined');
+
+export function partialFreeze<T>(obj: T): T {
```

Comment:
This is a bit messy, hoping you guys have a cleaner way to do this. Maybe modifying the original deepFreeze to do the same?

Reply:
> Maybe modifying the original deepFreeze to do the same?

💯% All users of `deepFreeze` should benefit from this goodness. There is already a [check](https://github.com/microsoft/vscode/blob/4a130c40ed876644ed8af2943809d08221375408/src/vs/base/common/objects.ts#L38) to decide to decent on an object or not. This check should be extended. 

Also, this shouldn't be checking for `Buffer` - that's a nodejs only type and where are here in  `common`-land. I also believe the error is coming from the underlying typed array, this fails the same way

```
> var a = new Uint8Array(12)
undefined
> Object.freeze(a)
Uncaught TypeError: Cannot freeze array buffer views with elements
    at Function.freeze (<anonymous>)
```

There should be a check iff an object is a typed array - quite a few candidates according to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Typed_arrays#typed_array_views. Good news is they are instanceof-able, e.g `candidate instanceof Float32Array` works etc.